### DISCUSSION TITLE
Juicefs-csi-driver limits

### DISCUSSION THREAD

**Discussion by sorend** [juicedata/juicefs#5205] [General]
Hi JuiceFS Community.

First: Thanks for an awesome product :-)

There is some data published regarding spec limits of JuiceFS itself:
https://juicefs.com/docs/cloud/reference/spec_limits/

As I investigate different CSI drivers, most of them come with some limits. For a few select it has been documented in the k8s documentation itself:
https://kubernetes.io/docs/concepts/storage/storage-limits/

Are there any spec limits for the juicefs-csi-driver?  (I would also be interested in someone who has run into actual real-world limits with the driver).

I am looking at operating a cluster with currently 1000+ persistent volumes across 20 nodes.

Thanks,
Soren

**Comment by zxh326** [comment]
No limit, depends on whether the node itself has enough resources, we will create different client pod(mountpods) for different pv.
